src/orion/core/io/database/mongodb.py

```diff
@@ -229,6 +229,9 @@ def count(self, collection_name, query=None):
 
         """
         dbcollection = self._db[collection_name]
+        if hasattr(dbcollection, 'count_documents'):
+            return dbcollection.count_documents(filter=query if query else {})
+
         return dbcollection.count(filter=query)
```

Comment:
this could be in an `else` statement, but that's a personal choice

Reply:
Actually pylint would complain because the else is useless. An alternative would be

```
if something:
    count = 
else:
    count = 
return count
```